app/models/client_notification.rb

```diff
@@ -0,0 +1,14 @@
+class ClientNotification < ApplicationRecord
+
+  before_save :ensure_exactly_one_active
+
+  default_scope { order(created_at: :desc) }
+  scope :active, -> { find_by(active: true) }
```

Comment:
Scopes are supposed to return chainable `ActiveRecord::Relation`s, in this case I think it would be better to have a class method (`def self.active ...`) (though scopes are actually just class methods too)